weave/trace_server/interface/builtin_object_classes/llm_structured_model.py

```diff
@@ -0,0 +1,51 @@
+from enum import Enum
+from typing import Any, Optional, Union
+
+from pydantic import BaseModel, Field
+
+import weave
+from weave.flow.model import Model
+from weave.trace_server.interface.builtin_object_classes import base_object_def
+
+
+class ResponseFormat(str, Enum):
+    JSON = "json"
+    TEXT = "text"
+
+    # TODO: Fast follow up
+    # JSON_SCHEMA = "jsonschema"
+
+
+class LLMStructuredCompletionModelDefaultParams(BaseModel):
+    temperature: Optional[float] = None
+    top_p: Optional[float] = None
+    max_tokens: Optional[int] = None
+    presence_penalty: Optional[float] = None
+    frequency_penalty: Optional[float] = None
+    stop: Optional[list[str]] = None
+    n_times: Optional[int] = None
+    response_format: Optional[ResponseFormat] = None
+    functions: Optional[list[dict]] = None
+
+
+class LLMStructuredCompletionModel(Model):
```

Comment:
Inheriting from Model will not work on the server since you can't import outside of `weave.trace_server` from here. This edges on the whole: how do we use the weave library in the server conversation